### PR TITLE
[stable10] Search api stable10

### PR DESCRIPTION
**Pull request by jvillafanez**
## Description
Backport of https://github.com/owncloud/core/pull/31873 and https://github.com/owncloud/core/pull/31943 to stable10

## Related Issue
https://github.com/owncloud/core/issues/25373

## How Has This Been Tested?
Added unittests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1:

File: apps/dav/lib/Capabilities.php [hunk 1]
```diff
@@ -30,7 +30,12 @@ public function getCapabilities() {
 		return [
 			'dav' => [
 				'chunking' => '1.0',
+				'zsync' => '1.0',
```
Comment (patrickjahns) [inline apps/dav/lib/Capabilities.php]: Did we backport zsync?

Comment (PVince81) [reply apps/dav/lib/Capabilities.php]: looks like a conflict resolution mistake. zsync is OC 11 only.

@jvillafanez can you fix ?

Comment (patrickjahns) [reply apps/dav/lib/Capabilities.php]: Send a fix via https://github.com/owncloud/core/pull/32603

Comment (PVince81) [reply apps/dav/lib/Capabilities.php]: ok already done in https://github.com/owncloud/core/pull/32603

**Pull request merged**